crates/ra_assists/src/handlers/replace_return_to_result.rs

```diff
@@ -0,0 +1,180 @@
+use ra_syntax::{
+    ast::{self, NameOwner, VisibilityOwner},
+    AstNode,
+    SyntaxKind::{
+        ATTR, COMMENT, CONST_DEF, ENUM_DEF, FN_DEF, MODULE, STRUCT_DEF, TRAIT_DEF, VISIBILITY,
+        WHITESPACE,
+    },
+    SyntaxNode, TextUnit, T,
+};
+
+use crate::{Assist, AssistCtx, AssistId};
+use ast::{make, Block, BlockExpr, Stmt};
+use test_utils::tested_by;
+
+// Assist: replace_return_to_result
+//
+// Swap the return function element to result
+//
+// ```
+// <|>fn foo() -> i32 { 42i32 }
+// ```
+// ->
+// ```
+// fn foo() -> Result<i32> { Ok(42i32) }
+// ```
+pub(crate) fn replace_return_to_result(ctx: AssistCtx) -> Option<Assist> {
+    let mut fn_def = ctx.find_node_at_offset::<ast::FnDef>();
+    if let Some(fn_def) = &mut fn_def {
+        if fn_def.ret_type().is_none() {
+            return None;
+        }
+        let ret_type = fn_def.ret_type().unwrap().type_ref();
+        if let Some(ret_type) = &ret_type {
+            if ret_type.syntax().text().to_string().starts_with("Result<") {
+                return None;
+            }
+        }
+        let mut exprs_to_wrap = vec![];
+        if let Some(fn_body) = &fn_def.body() {
+            if let Some(block_expr) = &fn_body.block() {
+                let statements = block_expr.statements();
+                for stmt in statements {
+                    let expr_stmt = match &stmt {
+                        ast::Stmt::ExprStmt(stmt) => stmt.expr(),
+                        _ => continue,
+                    };
+
+                    if let Some(expr) = &expr_stmt {
+                        match expr {
+                            ast::Expr::ReturnExpr(ret_expr) => {
+                                // Update return here
+                                if let Some(ret_expr_arg) = &ret_expr.expr() {
+                                    exprs_to_wrap.push(ret_expr_arg.syntax().clone());
+                                }
+                            }
+                            _ => continue,
+                        }
+                    };
+                }
+
+                collect_tail_exprs(&ctx, block_expr.clone(), &mut exprs_to_wrap);
+
+                let ret_type = ret_type.clone().unwrap();
+                return ctx.add_assist(
+                    AssistId("replace_return_to_result"),
+                    "Make Ok Wrapper",
+                    |edit| {
+                        for ret_expr_arg in exprs_to_wrap {
+                            edit.replace_node_and_indent(
+                                &ret_expr_arg,
+                                format!("Ok({})", ret_expr_arg),
+                            );
+                        }
+                        edit.target(ret_type.syntax().text_range());
+                        edit.replace_node_and_indent(
+                            ret_type.syntax(),
+                            format!("Result<{}>", ret_type),
```

Comment:
This is missing an error type.

Reply:
Yes good point ! I will put cursor on it to let user fill it